src/libsyntax/parse/parser.rs

```diff
@@ -5107,7 +5109,32 @@ impl<'a> Parser<'a> {
                     if has_parens {
                         self.expect(&token::CloseDelim(token::Paren))?;
                     }
-                    let poly_trait = PolyTraitRef::new(lifetime_defs, path, lo.to(self.prev_span));
+                    let poly_span = lo.to(self.prev_span);
```

Comment:
Looking at [the travis output](https://travis-ci.org/rust-lang/rust/jobs/475980590#L2085) it seems that you need to fix up the suggestion span:

```
20	LL | trait Tr3: !Sync + Unpin {} //~ ERROR negative trait bounds are not supported
-	   |            ^^^^^--
-	   |            |
-	   |            help: remove this trait bound
+	   |          --^^^^^
+	   |          |
+	   |          help: remove this trait bound
```

Reply:
my bad, I was leaving code and test quite out of sync. With this span, I was trying to leave the fixed code syntax-complete, thus I tried to remove the following `+` instead of the preceding `:`. In the general case, I imagine this will require much more info passed into this function.